aten/src/ATen/cuda/Atomic.cuh

```diff
@@ -330,7 +330,6 @@ inline __device__ void gpuAtomicAddNoReturn(int64_t *address, int64_t val) { gpu
 inline __device__ void gpuAtomicAddNoReturn(bool *address, bool val) { gpuAtomicAdd(address, val); }
 inline __device__ void gpuAtomicAddNoReturn(at::Half *address, at::Half val) { gpuAtomicAdd(address, val); }
 inline __device__ void gpuAtomicAddNoReturn(at::BFloat16 *address, at::BFloat16 val) { gpuAtomicAdd(address, val); }
-inline __device__ void gpuAtomicAddNoReturn(double *address, double val) { gpuAtomicAdd(address, val); }
 
 /* Special case fp32 atomic. */
```

Comment:
```suggestion
/* Note [HIP unsafeAtomicAdd]
 * ~~~~~~~~~~~~~~~~~~~~~~~~~~
 * Use unsafeAtomicAdd instead of atomicAdd for fp32 and fp64.
 * On HIP, atomicAdd is always a correct but slow CAS loop.
 * unsafeAtomicAdd will use HW instructions and is much faster,
 * but the caller must guarantee the pointer is GPU memory.
 * If the pointer is system memory, the result is a silent no-op.
 * This guarantee is upheld by all pytorch uses of unsafeAtomicAdd.
 */
```